devops/resources/run-app-permissions.sh

```diff
@@ -17,22 +17,53 @@ env
 assume-role ${ROLE_ARN}
 cd $WORKSPACE/configuration/playbooks
 
+# Function to set ASG and path variables
+function setASGAndPath () {
+	service_name="$1"
+	if [[ "${service_name}" == "lms" ]] || [[ "${service_name}" == "cms" ]] || [[ "${service_name}" == "edxapp" ]]; then
+		INVENTORY=$(./active_instances_in_asg.py --asg ${ENVIRONMENT}-${DEPLOYMENT}-worker)
+		SERVICE_ENV_PATH="/edx/app/edxapp/edxapp_env"
+		SERVICE_PYTHON_PATH="/edx/app/edxapp/venvs/edxapp/bin/python"
+		SERVICE_MANAGE_PATH="/edx/bin/manage.edxapp"
+	else
+		INVENTORY=$(./active_instances_in_asg.py --asg ${ENVIRONMENT}-${DEPLOYMENT}-${service_name})
+		SERVICE_ENV_PATH="/edx/app/${service_name}/${service_name}_env"
+		SERVICE_PYTHON_PATH="/edx/app/${service_name}/venvs/${service_name}/bin/python"
+		SERVICE_MANAGE_PATH="/edx/bin/manage.${service_name}"
+	fi
+}
+
 if [[ "$JOB_TYPE" =~ ^groups-(.+)$ ]]; then
 	service="${BASH_REMATCH[1]}"
 	configfile=${WORKSPACE}/app-permissions/groups/${service}.yml
-elif [[ "$JOB_TYPE" = 'active-users' ]] || [[ "$JOB_TYPE" = 'inactive-users' ]]; then
-	configfile=${WORKSPACE}/app-permissions/users/${ENVIRONMENT}-${DEPLOYMENT}.yml
+	setASGAndPath ${service}
+	if [[ "${service}" == "lms" ]] || [[ "${service}" == "cms" ]]; then
+		ANSIBLE_TAG="manage-$JOB_TYPE"
+	else
+		ANSIBLE_TAG="manage-groups-ida"
+	fi
+elif [[ "$JOB_TYPE" =~ ^(.+)-users-(.+)$ ]]; then
+	job_type_prefix="${BASH_REMATCH[1]}"
+	service="${BASH_REMATCH[2]}"
+	configfile=${WORKSPACE}/app-permissions/users/${service}/${ENVIRONMENT}-${DEPLOYMENT}.yml
+	setASGAndPath ${service}
+	if [[ "${service}" == "edxapp" ]]; then
+		ANSIBLE_TAG="manage-$JOB_TYPE"
+	else
+		ANSIBLE_TAG="manage-${job_type_prefix}-users-ida"
+	fi
 else
 	echo "Bad job type: ${JOB_TYPE}."
-	echo "Expected active-users, inactive-users, or groups-<service>."
+	echo "Expected active-users-edxapp, inactive-users-edxapp, inactive-users-<service> or groups-<service>."
 	exit 1
 fi
 
-INVENTORY=$(./active_instances_in_asg.py --asg ${ENVIRONMENT}-${DEPLOYMENT}-worker)
 if [[ -n ${INVENTORY} ]]; then
     ansible-playbook -i ${INVENTORY} manage_edxapp_users_and_groups.yml \
+										 -e "env_path=${SERVICE_ENV_PATH}" -e "python_path=${SERVICE_PYTHON_PATH}" \
+										 -e "manage_path=${SERVICE_MANAGE_PATH}" -e "service=${service}" \
                      -e@${configfile} -e "group_environment=${ENVIRONMENT}-${DEPLOYMENT}" \
-                     --user ${USER} --tags manage-${JOB_TYPE}
+                     --user ${USER} --tags ${ANSIBLE_TAG}
 else
     echo "Skipping ${ENVIRONMENT} ${DEPLOYMENT}, no worker cluster available, get it next time"
```

Comment:
While we're updating this script we may want to expand on this log message, what does it mean by get it next time?